packages/react/src/modules/feed/components/NotificationFeed/MarkAsRead.tsx

```diff
@@ -14,8 +15,10 @@ export const MarkAsRead: React.FC<MarkAsReadProps> = ({ onClick }) => {
   const { useFeedStore, feedClient, colorMode } = useKnockFeed();
   const { t } = useTranslations();
 
-  const unreadItems = useFeedStore((state) =>
-    state.items.filter((item) => !item.read_at),
+  // Without useShallow, the component will get itself in a re-render loop
+  // https://zustand.docs.pmnd.rs/migrations/migrating-to-v5#requiring-stable-selector-outputs
+  const unreadItems = useFeedStore(
+    useShallow((state) => state.items.filter((item) => !item.read_at)),
   );
```

Comment:
Could we refactor this to be:

```tsx
const {items} = useState(state => state.items);
const unreadItems = items.filter((item) => !item.read_at);
```

Maybe that needs to be in a useMemo for the filter? Not sure.

Reply:
the `useShallow` approach looks to be the proper way to access computed property based on state with the new Zustand: [https://zustand.docs.pmnd.rs/guides/prevent-rerenders-with-use-shallow](https://zustand.docs.pmnd.rs/guides/prevent-rerenders-with-use-shallow)

still want me to try `useMemo`?